pkg/executables/kubectl.go

```diff
@@ -1404,3 +1408,53 @@ func (k *Kubectl) GetResources(ctx context.Context, resourceType string, opts ..
 	stdOut, err := k.Execute(ctx, params...)
 	return stdOut.String(), err
 }
+
+// GetHardwareForCluster gets the hardwares with ownerName label.
+// Then iterates to filter the hardwares that correspond to the given clusterName.
+func (k *Kubectl) GetHardwareForCluster(ctx context.Context, clusterName, kubeconfigFile, namespace string) ([]tinkv1alpha1.Hardware, error) {
```

Comment:
This is also business logic. I would expect this method to exist on some other object within the tinkerbell provider and to simply call out to `Kubectl` to retrieve the Kubernetes objects.

Reply:
Sure. I can make this method into just `GetHardwareWithOwnerLabel` and add the filtering logic into a local method in the tinkerbell provider package.